### PR TITLE
Enhancements to the progress bars code

### PR DESCRIPTION
**Pull request by sarao1310**
- [ ] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [ ] This pull request is on the dev branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: utils/progressbar/transferprogressbarmanager.go [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+	go func() {
+		defer tpm.wg.Done()
+		for {
+			if tpm.shouldStop {
+				log.Info("phase 3, should stop is true")
+				return
+			}
+			ptr1, ptr2, _, _, err := getVals()
+			if err != nil {
+				log.Error(err)
```
Comment (eyalbe4) [inline utils/progressbar/transferprogressbarmanager.go:193]: Let's add a context perffix to this error message.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

**Comment by eyalbe4** [comment]
@sarao1310 - I see that there's an unresolved discussion above. Let me know when I can review this PR again.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
I see that this PR still has open discussions, so I'm returning it to you @sarao1310.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: utils/progressbar/progressbarmng.go [hunk 2]
```diff
@@ -68,6 +68,61 @@ func NewBarsMng() (mng *ProgressBarMng, shouldInit bool, err error) {
...
+func (bm *ProgressBarMng) newHeadlineTaskProg(getVal func() (ptr1, ptr2 *int64, err error), headLine, valHeadLine string) *TasksWithHeadlineProg {
+	bm.barsWg.Add(1)
+	prog := TasksWithHeadlineProg{}
+	prog.headlineBar = bm.NewHeadlineBar(headLine)
+	prog.tasksProgressBar = bm.newTasksProgBar(getVal, valHeadLine)
+	prog.emptyLine = bm.NewHeadlineBar("")
+	return &prog
+}
+
 func (bm *ProgressBarMng) NewTasksWithHeadlineProg(totalTasks int64, headline string, spinner bool, color Color, windows bool, taskType string) *TasksWithHeadlineProg {
```
[the remark this answers is not in the document]
Comment (sarao1310) [reply utils/progressbar/progressbarmng.go:127]: Done

**Pull request merged**